File: Algorithm/go/blind75/252-MeetingRooms.go
package main

import "sort"

func canAttendMeetings(intervals [][]int) bool {
	if len(intervals) < 2 {
		return true
	}
	sort.Slice(intervals, func(i, j int) bool {
		return intervals[i][0] < intervals[j][0]
	})

	for i := 0; i < len(intervals)-1; i++ {
		if intervals[i+1][0] < intervals[i][1] {
			return false
		}
	}
	return true
}

File: Algorithm/python/Blind75/252-MeetingRooms.py
from typing import List


class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        # sort meeting start time
        intervals = sorted(intervals, key=lambda x: x[0])

        for i in range(1, len(intervals)):
            if intervals[i - 1][1] > intervals[i][0]:
                return False

        return True
